source/common/rate_limiter_impl.cc

```diff
@@ -118,4 +160,21 @@ bool FilteringRateLimiterImpl::tryAcquireOne() {
   return rate_limiter_->tryAcquireOne() ? filter_() : false;
 }
 
+GraduallyOpeningRateLimiterFilter::GraduallyOpeningRateLimiterFilter(
```

Comment:
(fyi) Now that I read the code and understand the implementation here, I have to say that this is truly beautiful, including your use of class inheritance for the ForwardingRateLimiter.

I have learned something today and I appreciate it, thanks.

Reply:
That is nice to hear, thanks!